rootfs/api/models/app.py

```diff
@@ -385,6 +400,79 @@ def _default_structure(self, release):
 
         return structure
 
+    def verify_application_health(self, **kwargs):
+        """
+        Verify an application is healthy via the router.
+        This is only used in conjunction with the kubernetes health check system and should
+        only run after kubernetes has reported all pods as healthy
+        """
+        # Bail out early if the application is not routable
+        if not kwargs.get('routable', False):
+            return
+
+        app_type = kwargs.get('app_type')
+        self.log(
+            'Waiting for router to be ready to serve traffic to process type {}'.format(app_type),
+            level=logging.DEBUG
+        )
+
+        # Get the router host and append healthcheck path
+        url = 'http://{}:{}'.format(settings.ROUTER_HOST, settings.ROUTER_PORT)
```

Comment:
> I thought we were aiming to not have the controller talk directly to the router

I think it's with respect to implementation-specifics that we don't want these two components communicating.  Since the controller is only making a request to the application, passing through the router in the process, much as any actual end-client would, I think this is ok.

Having said that, @arschles and I have traded a lot of ideas back and forth about how to make all the nodes of the router mesh more immediately consistent with one another, with the ability to report back to any interested party about the current state of its configuration... however, _that_ would involve some implementation-specifics, and I think @arschles and I agreed we'd hold off on trying to solve this difficult problem until post-stable.

For the time being, I think what @helgi has done the right thing.